src/code/z_camera.c

```diff
@@ -7169,21 +7175,18 @@ s32 Camera_CheckWater(Camera* camera) {
             if (*quakeId != 0) {
                 Quake_RemoveFromIdx(*quakeId);
             }
-            camera->unk_150 = 0;
-            camera->unk_152 = 0;
+            camera->waterDistortionTimer = 0;
+            camera->distortionFlags = 0;
         }
         Audio_SetExtraFilter(0);
     }
     //! @bug: doesn't always return a value, but sometimes does.
 }
 
-/**
- * Sets the room to be hot camera quake flag
- */
-s32 Camera_SetRoomHotFlag(Camera* camera) {
-    camera->unk_152 &= ~1;
+s32 Camera_SetHotRoomDistortionFlag(Camera* camera) {
+    camera->distortionFlags &= ~DISTORTION_HOT_ROOM;
     if (camera->globalCtx->roomCtx.curRoom.unk_02 == 3) {
-        camera->unk_152 |= 1;
+        camera->distortionFlags |= DISTORTION_HOT_ROOM;
     }
```

Comment:
does the flag being on enable the distortion in general? feel like the function can be less specific about what its doing and say `EnableHotRoomDistortion`
the fact that it sets a flag to do that is a more of an implementation detail

Reply:
I think "enable" makes it sound too much like it will always enable instead of choosing to enable or not?

since this is called in camera_update what about `Camera_UpdateHotRoomDistortionState`
maybe "state" is good enough to abstract away the fact it's a flag?

Reply:
This check is run every frame in camera_update, both the function that sets the flag and the function that calculates the distortion. So by setting the `distortionFlags` (either here or anywhere else, any actor could in theory grab the active camera and set its distortion flags), that will lead to the distortion being applied in the update function. So yes, I could call distortion general based on your context. The code in `camera_update` that sets the distortion flags are:
```
if (camera->status == CAM_STAT_ACTIVE) {
    Camera_CheckWater(camera);
    Camera_SetHotRoomDistortionFlag(camera);
}
```
Where part of `Camera_CheckWater` sets the distortion flags for entering water (but does many other things as well).

So alternatively to parallel that, what about:
```
if (camera->status == CAM_STAT_ACTIVE) {
    Camera_CheckWater(camera);
    Camera_CheckHotRoom(camera);
}
```
So the name `Camera_CheckHotRoom` checks to see if current room is a hot room, and updates whatever parameters it needs to (which in this case is the distortion flag). But also, that might be too general since it does lose the `distortion` part of the function name

Reply:
been a while since we talked about this so I may forget some context, but yeah "CheckHotRoom" might be a bit too general for what I understand it to do

an argument could be made that it can be `UpdateHotRoom` and `UpdateWater`. the water one does more than just the distortion, and in theory you could make the camera do more than just distortion in a hot room

Reply:
These are both run inside `Camera_Update` so I would actually be fine with both `UpdateHotRoom` and `UpdateWater`

Reply:
Actually `Camera_CheckWater` is also run in an init function (specifically `Camera_InitPlayerSettings`) so I'm a little less sure about using `Update` instead of `Check`

Reply:
I would have to read more of the code to give a better answer, but sometimes it does make sense to "update" something in init, if you want it to run unconditionally when it first starts. or have it be updated before the first update cycle starts